tfjs-layers/src/layers/preprocessing/random_width.ts

```diff
@@ -0,0 +1,160 @@
+  private widthFactor: Tensor<Rank.R3>|Tensor<Rank.R4>;
+
+  constructor(args: RandomWidthArgs) {
+    super(args);
+    this.factor = args.factor;
+
+    if (Array.isArray(this.factor) && this.factor.length === 2) {
+      this.widthLower = this.factor[0];
+      this.widthUpper = this.factor[1];
+    } else if (!Array.isArray(this.factor) && this.factor > 0){
+      this.widthLower = -this.factor;
+      this.widthUpper = this.factor;
+    } else {
+      throw new ValueError(
+        `Invalid factor: ${this.factor}. Must be positive number or tuple of 2 numbers`
+      );
+    }
+    if (this.widthLower < -1.0 || this.widthUpper < -1.0) {
+      throw new ValueError(
+        `factor must have values larger than -1. Got: ${this.factor}`
+      );
+    }
+
+    if (this.widthUpper < this.widthLower) {
+      throw new ValueError(
+        `factor cannot have upper bound less than lower bound.
+        Got upper bound: ${this.widthUpper}.
+        Got lower bound: ${this.widthLower}
+      `);
+    }
+
+    if(args.seed) {
+      this.seed = args.seed;
+    } else {
+      this.seed = null;
+    }
+
+    if (args.interpolation) {
+      if (INTERPOLATION_METHODS.has(args.interpolation)) {
+        this.interpolation = args.interpolation;
+      } else {
+        throw new ValueError(`Invalid interpolation parameter: ${
+            args.interpolation} is not implemented`);
+      }
+    } else {
+      this.interpolation = 'bilinear';
+    }
+  }
+
+  override getConfig(): serialization.ConfigDict {
+    const config: serialization.ConfigDict = {
+      'factor': this.factor,
+      'interpolation': this.interpolation,
+      'seed': this.seed,
+    };
+
+    const baseConfig = super.getConfig();
+    Object.assign(config, baseConfig);
+    return config;
+  }
+
+  override computeOutputShape(inputShape: Shape|Shape[]): Shape|Shape[] {
+    inputShape = getExactlyOneShape(inputShape);
+    const numChannels = inputShape[2];
+    return [this.imgHeight, this.adjustedWidth, numChannels];
+  }
+
+  override call(inputs: Tensor<Rank.R3>|Tensor<Rank.R4>,
+    kwargs: Kwargs): Tensor[]|Tensor {
+
+    return tidy(() => {
+        const input = getExactlyOneTensor(inputs);
+        const inputShape = input.shape;
+        this.imgHeight = inputShape[inputShape.length - 3];
+        const imgWidth = inputShape[inputShape.length - 2];
+
+        const randomUniform = this.setRNGType('uniform');
+
+        if (this.seed !== null) {
+          this.widthFactor = randomUniform([1],
+            (1.0 + this.widthLower), (1.0 + this.widthUpper),
+            'float32', this.seed
+          );
+        } else {
+          this.widthFactor = randomUniform([1],
+            (1.0 + this.widthLower), (1.0 + this.widthUpper)
+          );
+        }
+        this.adjustedWidth = this.widthFactor.dataSync()[0] * imgWidth;
+        this.adjustedWidth = Math.round(this.adjustedWidth);
+        const size: [number, number] = [this.imgHeight, this.adjustedWidth];
```

Comment:
Tiny nit: Using `as const` makes TypeScript correctly infer this as `const [number, number]` instead of `number[]`.
```suggestion
        const size = [this.imgHeight, this.adjustedWidth] as const;
```

Reply:
resizeBilinear and resizeNearestNeighbor expect a mutable array of two numbers. using  `as const` assigns the readonly type